src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ImportCompletionItem.cs

```diff
@@ -1,51 +1,68 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
+using System.Collections.Immutable;
 using System.Diagnostics;
-using System.Globalization;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.Completion.Providers
 {
-    internal static class TypeImportCompletionItem
+    internal static class ImportCompletionItem
     {
         private const string SortTextFormat = "~{0} {1}";
-        private const string GenericTypeNameManglingString = "`";
-        private static readonly string[] s_aritySuffixesOneToNine = { "`1", "`2", "`3", "`4", "`5", "`6", "`7", "`8", "`9" };
 
         private const string TypeAritySuffixName = nameof(TypeAritySuffixName);
         private const string AttributeFullName = nameof(AttributeFullName);
+        private const string SymbolKeyData = nameof(SymbolKeyData);
 
         public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containingNamespace, string genericTypeSuffix)
         {
-            PooledDictionary<string, string> propertyBuilder = null;
+            return Create(typeSymbol.Name, typeSymbol.Arity, containingNamespace, typeSymbol.GetGlyph(), genericTypeSuffix, CompletionItemFlags.CachedAndExpanded, symbolKeyData: null);
+        }
 
-            if (typeSymbol.Arity > 0)
+        public static CompletionItem Create(string name, int arity, string containingNamespace, Glyph glyph, string genericTypeSuffix, CompletionItemFlags flags, string? symbolKeyData)
+        {
+            ImmutableDictionary<string, string>? properties = null;
+
+            if (symbolKeyData != null || arity > 0)
             {
-                propertyBuilder = PooledDictionary<string, string>.GetInstance();
-                propertyBuilder.Add(TypeAritySuffixName, GetAritySuffix(typeSymbol.Arity));
+                var builder = PooledDictionary<string, string>.GetInstance();
+
+                if (symbolKeyData != null)
+                {
+                    builder.Add(SymbolKeyData, symbolKeyData);
+                }
+                else
+                {
+                    builder.Add(TypeAritySuffixName, AbstractDeclaredSymbolInfoFactoryService.GetMetadataAritySuffix(arity));
+                }
+
+                properties = builder.ToImmutableDictionaryAndFree();
```

Comment:
what if you have symbolKeyData *and* arity?

Reply:
if that can't happen, then add assert it can't happen.

Reply:
It can happen, but we wouldn't need arity to recover symbol when we already have SymbolKey id.

Reply:
doc this plz :)

Reply:
Done